concepts/observability-configurable-logging/07-multiple-logpipeline-investigation.md

```diff
@@ -0,0 +1,133 @@
+2. Two log pipelines:
+   - [Loki](./assets/logpipeline-investigation/setup-3c/loki.yml)
+   - [Http output](./assets/logpipeline-investigation/setup-3c/mock-server.yml)
+
+![a](./assets/logpipeline-investigation/setup-3c/setup-3c.drawio.svg)
+
+Result:
+- Http output was having an outage. The Loki output kept working.
+- The chunks in the `filesystem buffer` are rolled: Old chunks are discarded; new ones created.
+- When Loki output was having an outage, the http output kept working.
+
+![a-down](./assets/logpipeline-investigation/setup-3c/dashboard-loki-down.png)
+
+
+![loki-down](./assets/logpipeline-investigation/setup-3c/dashboard-loki-down.png)
+
+### Setup 3d
+Setup:
+1. One input with two outputs (Loki (Loki plugin), and http output (http plugin)) without rewrite tags
+2. Two log pipelines:
+   - [Loki](./assets/logpipeline-investigation/setup-3d/loki.yml)
+   - [Http output](./assets/logpipeline-investigation/setup-3d/mock-server.yml)
+
+![a](./assets/logpipeline-investigation/setup-3d/setup-3d.drawio.svg)
+
+Result:
+- Http output was having an outage. This led to the filling of the tail filesystem buffer.
+- After the tail plugin filesystem buffer was full, it kept reading.
+- The chunks in the tail plugin filesystem buffer are rolled: Old chunks are discarded; new ones created.
+- Loki output stopped working as well.
+- The logs are stored in filesystem buffer before applying the kuberenets filter. There is a risk of not being able to fetch the kubernetes metadata of the logs. This applies if the pod has been terminated before the logs have been read from the filesystem again for next stage of pipeline processing.
+
+
+A known [Fluent Bit GitHub issue](https://github.com/fluent/fluent-bit/issues/4373) describes the same problem.
+
+### Setup 4
+![a](./assets/logpipeline-investigation/setup-4/setup-4.drawio.svg)
+
+Setup:
+1. One input with two outputs (limiting the maximum number of chunks in the `filesystem` buffer) without rewrite tags
+2. Two log pipelines:
+   - [Loki](./assets/logpipeline-investigation/setup-4/loki.yaml)
+   - [Http output](./assets/logpipeline-investigation/setup-4/mockserver.yml)
+
+Result:
+- When one of the outputs is having an outage, the filesystem buffer is filled up and only keeps 150M of latest data (old data is discarded first). Eventually, when the filesystem buffer at tail plugin is full, the output plugin for Loki stops as well.
+- When both outputs are having an outage, the tail plugin is stopped after filling the filesystem buffer, even though the buffer was not completely full (104M/150M).
+
+## Summary
+We performed various tests and found that with filesystem buffering (through rewrite_tag filter) is necessary to prevent loss of logs when fluent-bit pod is restarted. Additionally when one of output is having an outage, then the logs are still streamed to the other output. However loss of logs cannot be prevented completely, if the buffer is filled up then fluentbit keeps only the latest logs.
```

Comment:
I'm not completely sure about the meaning of the first sentence. We need a comma, and maybe an "it". The meanings will be different so I don't know which one is correct.

```suggestion
We performed various tests and found that with filesystem, buffering (through rewrite_tag filter) is necessary to prevent loss of logs when fluent-bit pod is restarted. Additionally, when one of the outputs is having an outage, the logs are still streamed to the other output. However, loss of logs cannot be prevented completely: If the buffer is filled up, Fluent Bit keeps only the latest logs.
```

or 



```suggestion
We performed various tests and found that with filesystem buffering (through rewrite_tag filter), it is necessary to prevent loss of logs when fluent-bit pod is restarted. Additionally, when one of the outputs is having an outage, the logs are still streamed to the other output. However, loss of logs cannot be prevented completely: If the buffer is filled up, Fluent Bit keeps only the latest logs.
```